### ISSUE TITLE
Update hyper to 0.12

### ISSUE THREAD

**Issue by SimonSapin** [servo/servo#19825]
This seems to be blocked on https://github.com/servo/servo/issues/18518 at the moment.

It also blocks updating url to a for-now-hypothetical 2.0 version.

(Filing this tracking issue so that I can find this info more easily next time.)

**Comment by Eijebong** [comment]
Just to put every link there:

- https://github.com/cyderize/rust-websocket/issues/132
- https://github.com/hyperium/hyper/pull/1287
- https://github.com/hyperium/hyper/issues/1323
- https://github.com/hyperium/hyper/issues/1395

**Comment by jdm** [comment]
fyi @avadacatavra

**Comment by Eijebong** [comment]
Also, https://github.com/nox/hyper_serde/pull/20 this one is already done

**Comment by jdm** [comment]
https://github.com/hyperium/hyper/issues/1395 is tracking the missing protocol upgrade work that nobody has done.

**Comment by Eijebong** [comment]
Done now :)
